### ISSUE TITLE
Seemingly unrelated CB: in solo for DropSeq

### ISSUE THREAD

**Issue by ktpolanski** [alexdobin/STAR#1211]
Hello, long-time user (even prior to solo), big fan of the tool. I now get to use solo to process 10X quickly, and all has been going great.

However, a colleague recently received some DropSeq data from a collaborator. I mapped it with 2.7.3a using the following parameterisation, to be as consistent as possible with our aforementioned 10X processing:

        ~/STAR-2.7.3a/bin/Linux_x86_64/STAR --runThreadN `grep -c ^processor /proc/cpuinfo` \
                --genomeDir ~/STAR-2.7.3a/GRCh38-3.0.0 \
                --readFilesIn ../$SAMPLE\_2.fastq.gz ../$SAMPLE\_1.fastq.gz \
                --readFilesCommand zcat \
                --outSAMtype BAM SortedByCoordinate \
                --outSAMmultNmax 1 \
                --outMultimapperOrder Random \
                --runRNGseed 1 \
                --outSAMattributes NH HI AS nM CB UB GX GN \
                --soloType CB_UMI_Simple \
                --soloCBwhitelist None \
                --soloBarcodeReadLength 0 \
                --soloCBstart 1 \
                --soloCBlen 12 \
                --soloUMIstart 13 \
                --soloUMIlen 8 \
                --soloUMIfiltering MultiGeneUMI \
                --soloCellFilter CellRanger2.2 3000 0.99 10 \
                --soloFeatures Gene GeneFull Velocyto \
                --soloOutFileNames logs/ genes.tsv barcodes.tsv matrix.mtx

The colleague then tried to use the BAM and `filtered/barcodes.tsv` for downstream use, and came back saying that he was encountering errors and very poor results. I went to investigate, and it would appear that for whatever reason `CB:` is quite unrelated to what was provided on input.

To elaborate, here are a few R1s:
```
@SRR11402938.1.226895 226895 length=26
GCCCGGGTCATCCGAACATCCTTTTA
+SRR11402938.1.226895 226895 length=26
FFFFFFFFFFFFFFFFFFFFFFF,,,
@SRR11402938.1.5503949 5503949 length=26
TTTCGCTAGTTACTGGGAATGGGGGG
+SRR11402938.1.5503949 5503949 length=26
F:FFFFFFFFFFFFFFFFFFF,,,F,
```

Here are their corresponding BAM entries. The `UB:` matches what appears from base 13 onward in `R1`, but the `CB:` seems to have nothing to do with the first 12 bases.
```
SRR11402938.1.226895    16      1       14404   0       21S55M  *       0       0       TTTTTTTTTTTTTTTTTTTTTTTTCTGCTCAGTTCTTTATTGATTGGTGTGCCGTTTTCTCTGGAAGCCTCTTAAG       FFF,FFFFFFFFFFFFFFF,F:FFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF    NH:i:8  HI:i:1  AS:i:54 nM:i:0     CB:Z:CTGGGCTCATAT       UB:Z:CGAACATC
SRR11402938.1.5503949   16      1       14423   1       76M     *       0       0       TGATTGGTGTGCCGTTTTCTCTGGAAGCCTCTTAAGAACACAGTGGCGCAGGCTGGGTGGAGCCGTCCCCCCATGG       FFFFFFFFFFFFFFF:FFFFFFFFFFFF:FFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFF,FFFFFFF    NH:i:4  HI:i:1  AS:i:74 nM:i:0     CB:Z:TAGTCGCCTCGC       UB:Z:CTGGGAAT
```

Out of pure curiosity, I extracted `GCCCGGGTCATC`, the first twelve bases of the first of those reads, and managed to find it as part of `filtered/barcodes.tsv`. So according to that, it should be in the top 4510 barcodes for the sample. I then ran `samtools view Aligned.sortedByCoord.out.bam | grep "CB:Z:GCCCGGGTCATC" | wc -l`, which yielded 42. Checking 42 against `UMIperCellSorted.txt` puts it tied for 116,447th place.

So in summary - it seems that for whatever reason, for this particular mapping, the `CB:` is unrelated to the information in `R1`, and also fails to agree with `filtered/barcodes.tsv`. Some assistance please? Thanks and sorry for the trouble.

**Comment by ktpolanski** [comment]
Seeing how the only meaningful difference in parameterisation between my known-functional 10X processing and this is `--soloCBwhitelist None`, I took the filtered barcodes from the borked BAM run and used them as the whitelist. Suddenly the BAM has sensible information in `CB:` again:

```
SRR11402938.1.226895    16      1       14404   0       21S55M  *       0       0       TTTTTTTTTTTTTTTTTTTTTTTTCTGCTCAGTTCTTTATTGATTGGTGTGCCGTTTTCTCTGGAAGCCTCTTAAG       FFF,FFFFFFFFFFFFFFF,F:FFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF    NH:i:8  HI:i:1  AS:i:54 nM:i:0     CB:Z:GCCCGGGTCATC       UB:Z:CGAACATC
```

While this resolution makes the situation less pressing, it nevertheless remains a bug I encountered. Seems extremely likely that it's related to `--soloCBwhitelist None`.
